source/aspnetintegration.rst

```diff
@@ -263,3 +266,210 @@ A typical solution to this problem is to split up the class into multiple smalle
+        //constructor may be called before all of the registrations have been added to the container.
+        public SimpleInjectorJobProcessorHostedService(Container container) => _container = container;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            /*
+             * Now the adapter can resolve things from the container.
+             * For example, we could get a settings object from the container:
+             * ISettingsObj settings = _containter.GetInstance<ISettingsObj>();
+             * _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(settings.TimerInterval));
+             */
+            _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
+        {
+            // Run operation in a scope
+            using (AsyncScopedLifestyle.BeginScope(_container))
+            {
+                // Resolve the collection of IMyJob application-specific implementations and run them all
+                foreach (var service in _container.GetAllInstances<IMyJob>())
+                {
+                    service.DoWork();
+                }
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _timer?.Change(Timeout.Infinite, 0);
+            return Task.CompletedTask;
+        }
+
+        public void Dispose() => _timer?.Dispose();
+    }
+
+Then the IHostedService adapter would get registered in the Startup class in the following way (for the rest of the Simple Injector-specific code see the Startup snippet above):
+
+.. code-block:: c#
+
+    public class Startup
+    {
+        private Container container = new Container();
+
+        //...Other startup initialization code here...
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // ASP.NET default stuff here
+
+            services.AddSingleton<IHostedService>(new SimpleInjectorJobProcessorHostedService(container));
+
+            IntegrateSimpleInjector(services);
+        }
+
+        //...Configuration code here...
+    }
+
+This approach allows the application-specific code to remain oblivious of ASP.NET's implementation details, such as IHostedService.  For example, an IMyJob implementation might look like this:
+
+.. code-block:: c#
+
+    public class DoSomethingJob : IMyJob
+    {
+        private readonly IJobRepository _repo;
+        private readonly ILogger _logger;
+
+        //No knowledge of ASP.NET Core dependencies is needed.
+        public DoSomethingJob(IJobRepository repo, ILogger logger)
+        {
+            _repo = repo;
+            _logger = logger;
+        }
+
+        public void DoWork() => ...do something here...
+    }
+
+The IMyJob implementations can be registered with SimpleInjector as usual:
```

Comment:
Typo: It's "Simple Injector", not "SimpleInjector".

Reply:
Sorry about that.  I did a quick search through the document and made sure that it didn't appear like that in any of the other text that I wrote.